src/Accelerators/NNPA/Conversion/ONNXToZHigh/RewriteONNXForZHigh.cpp

```diff
@@ -388,6 +388,60 @@ class SplitLargeMatMulPattern : public OpRewritePattern<ONNXMatMulOp> {
   }
 };
 
+/// This pattern is to replace `C = add/sub(A, B)` by `A` when B is a zero
+/// defined by ConstantOfShape and C's shape is the same as A's shape.
+/// In other words, the output does not depend on the second operand.
+template <typename OP_TYPE>
+class AddSubWithZeroConstantOfShapePattern : public OpRewritePattern<OP_TYPE> {
```

Comment:
It's better to do this pattern at ONNX level, but since it uses DimAnalysis and we have no onnx-to-onnx pass that uses DimAnalysis, so we put this pattern here.